src/Components/Theme/theme.ts

```diff
@@ -146,102 +140,44 @@ const changeTheme = async (theme?: string, category?: '*' | 'root' | 'tabbing' |
 			const styleKey = splittedKey[splittedKey.length - 1];
 			if (key.startsWith('hljs')) {
 				const className = formalKey.split('.').slice(0, -1).join('.').replace('hljs.', 'hljs-');
-				styles += `.${className} { ${styleKey}: ${value}; }\n`;
+				content += `.${className} { ${styleKey}: ${value} }\n`;
 			} else {
-				for (const _category of [
-					'root',
-					'windowmanager',
-					'tabs',
-					'settings',
-					'favorites',
-					'grid',
-					'contextmenu',
-					'prompt',
-					'preview',
-					'properties',
-				]) {
-					if (key.startsWith(_category)) {
-						const usingClassName = formalKey[_category.length] === '.';
-						const className = formalKey.split('.').slice(1, -1).join('.');
-						const idName = formalKey.split('#').slice(1).join('#').split('.')[0];
-						for (const allowed_style of ALLOWED_STYLES) {
-							if (styleKey.startsWith(allowed_style)) {
-								if (styleKey.startsWith('--')) {
-									styles += `:root { ${styleKey}: ${value}; }\n`;
-								} else {
-									styles += `${
-										usingClassName
-											? className === ''
-												? _category === 'root'
-													? ':root'
-													: '.' + _category
-												: '.' + className
-											: '#' + idName
-									}{ ${styleKey}: ${value}; }\n`;
-								}
-								break;
-							}
+				for (const _category of CATEGORIES) {
+					if (!key.startsWith(_category)) continue;
+					const usingClassName = formalKey[_category.length] === '.';
+					const className = formalKey.split('.').slice(1, -1).join('.');
+					const idName = formalKey.split('#').slice(1).join('#').split('.')[0];
+					for (const allowedStyle of ALLOWED_STYLES) {
+						if (!styleKey.startsWith(allowedStyle)) continue;
+						if (styleKey.startsWith('--')) {
+							content += `:root { ${styleKey}: ${value}; }\n`;
+							break;
 						}
+						if (!usingClassName) content += '#' + idName;
+						else if (className !== '') content += '.' + className;
+						else if (_category !== 'root') content += '.' + _category;
+						else content += ':root';
+						content += `{ ${styleKey}: ${value} }\n`;
+						break;
 					}
 				}
 			}
 		}
-		style.innerHTML = styles;
-		if (!document.head.contains(style)) document.head.appendChild(style);
+		style.textContent = content;
```

Comment:
Oh and the reason I use `textContent` instead of `innerHTML` is that it's more efficient on performance.